src/monmove.cpp

```diff
@@ -277,10 +279,57 @@ float monster::rate_target( Creature &c, float best, bool smart ) const
     return INT_MAX;
 }
 
+bool monster::is_item_lootable( const item &itm )
+{
+    bool lootable = false;
+
+    //TODO: Definitely find a better way to do the below, too much repeatability
+
+    if( !lootable_categories.empty() )
+        for( std::string tag : lootable_categories ) {
+            if( itm.get_category().name() == tag ) {
+                lootable = true;
+            } else {
+                if( lootables_requires_all ) {
+                    return false;
+                }
+            }
+        }
+
+    if( !lootable_materials.empty() )
+        for( std::string tag : lootable_materials ) {
+            if( itm.made_of( material_id( tag ) ) ) {
```

Comment:
Instead of casting it here, make `lootable_materials` contain instances of `material_id` instead of strings.